src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityService.kt

```diff
@@ -211,7 +211,7 @@ class ActivityService(
       agencyInternalLocationRepository.findByIdOrNull(it)
         ?: throw BadDataException("Location with id=$it does not exist")
     }?.also {
-      if (it.agencyId != prisonId) {
+      if (it.agency.id != prisonId) {
```

Comment:
An AgencyInternalLocation now has a parent prison, location and ref data rather than just the codes